packages/hdwallet-native/src/binance.ts

```diff
@@ -88,7 +89,8 @@ export function MixinNativeBinanceWallet<TBase extends core.Constructor<NativeHD
           throw Error("Invalid permissions to sign for address")
         }
         const addressTo = msg.tx.msgs[0].outputs[0].address;
-        const amount = (msg.tx.msgs[0].inputs[0].coins[0].amount).toString();
+        // The Binance SDK takes amounts as decimal strings.
+        const amount = new BigNumber(msg.tx.msgs[0].inputs[0].coins[0].amount).shiftedBy(-8).toString();
```

Comment:
do we need an `isNaN` check here ?

Reply:
yes we do. I misread the docs on `shiftedBy()` and thought it would throw.

Reply:
added an `isInteger()` check before the shift, just for good measure.

Reply:
that will also make sure anything feeding in a decimal -- like it would have had to do to work around this bug -- will fail fast.